app/models/agents/pushbullet_agent.rb

```diff
@@ -25,14 +37,39 @@ def default_options
       {
         'api_key' => '',
         'device_id' => '',
-        'title' => "Hello from Huginn!",
+        'title' => "{{title}}",
         'body' => '{{body}}',
+        'type' => 'note',
       }
     end
 
+    form_configurable :api_key, roles: :validatable
+    form_configurable :device_id, roles: :completable
+    form_configurable :type, type: :array, values: ['note', 'link', 'address']
+    form_configurable :title
+    form_configurable :body, type: :text
+    form_configurable :url
+    form_configurable :name
+    form_configurable :address
+
     def validate_options
-      errors.add(:base, "you need to specify a pushbullet api_key") unless options['api_key'].present?
-      errors.add(:base, "you need to specify a device_id") if options['device_id'].blank?
+      errors.add(:base, "you need to specify a pushbullet api_key") if options['api_key'].blank?
+      create_device if options['device_id'].blank?
```

Comment:
Bit strange to make changes during validation.  Any other way to do this, or does it at least avoid making more than one?